azure-iot-device/azure/iot/device/iothub/models/methods.py

```diff
@@ -0,0 +1,70 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains classes related to direct method invocations.
+"""
+
+
+class MethodRequest(object):
+    """Represents a request to invoke a direct method.
+
+    :ivar str request_id: The request id.
+    :ivar str name: The name of the method to be invoked.
+    :ivar dict payload: The JSON payload being sent with the request.
+    """
+
+    def __init__(self, request_id, name, payload):
+        """Initializer for a MethodRequest.
+
+        :param str request_id: The request id.
+        :param str name: The name of the method to be invoked
+        :param dict payload: The JSON payload being sent with the request.
+        """
+        self._request_id = request_id
+        self._name = name
+        self._payload = payload
+
+    @property
+    def request_id(self):
+        return self._request_id
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def payload(self):
+        return self._payload
+
+
+class MethodResponse(object):
+    """Represents a response to a direct method.
+
+    :ivar str request_id: The request id of the MethodRequest being responded to.
+    :ivar int status: The status of the execution of the MethodRequest.
+    :ivar dict payload: The JSON payload to be sent with the response.
+    """
+
+    def __init__(self, request_id, status, payload=None):
+        """Initializer for MethodResponse.
+
+        :param str request_id: The request id of the MethodRequest being responded to.
+        :param int status: The status of the execution of the MethodRequest.
+        :param dict payload: The JSON payload to be sent with the response. (OPTIONAL)
```

Comment:
>dict [](start = 15, length = 4)

json != dict.  json can also be an array, string, number, "true", "false", or "null".  ref: json.org

Reply:
Yeah, sometimes it wouldn't be a dictionary. The type is whatever the result of json.loads is. I'll correct the docs

---
In reply to: [282743221](https://github.com/Azure/azure-iot-sdk-python-preview/pull/86#discussion_r282743221) [](ancestors = 282743221)